### PR TITLE
ADEN-7004 Adds AdEngine assets

### PR DESCRIPTION
**Pull request by leshniak**
## Links

* http://wikia-inc.atlassian.net/browse/ADEN-7004

## Description

Adds AdEngine assets optionally and asynchronously.

## Reviewers

@hakubo  @fraszczakszymon

### CODE REVIEW

File: app/modules/adengine-assets.js [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+			styles.rel = 'stylesheet';
+			styles.href = assetUrls.adProductsStyles;
+			document.head.appendChild(styles);
+
+			this.wasStyleLoadInitialized = true;
+		}
+	}
+
+	loadScript(url) {
+		return new Promise((resolve) => {
```
Comment (hakubo) [inline app/modules/adengine-assets.js]: we also have $script to load scripts which is I think more robust

File: app/modules/adengine-assets.js [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+	}
+
+	load() {
+		this.loadStyles();
+
+		return this.loadScripts();
+	}
+}
+
+export default new AdEngineAssets();
```
Comment (hakubo) [inline app/modules/adengine-assets.js]: not sure if exporting an instance is a good practice
I'd say if you need this as a singleton - just use module - you don't need class for that

File: app/modules/adengine-assets.js [hunk 3]
```diff
@@ -0,0 +1,61 @@
+import {Promise} from 'rsvp';
+
+export const assetUrls = {
+	adEngineScript: '/mobile-wiki/assets/adengine/ad-engine.global.js',
+	adProductsScript: '/mobile-wiki/assets/adengine/ad-products.global.js',
+	geoScript: '/mobile-wiki/assets/adengine/geo.global.js',
+	adProductsStyles: '/mobile-wiki/assets/adengine/ad-products.css'
```
Comment (fraszczakszymon) [inline app/modules/adengine-assets.js]: We already have styles added:
https://github.com/Wikia/mobile-wiki/blob/d22c6623abd692fe45453721c4730d4d6a653f72/app/styles/module/_ads.scss#L4

We can skip this part in this place

Comment (leshniak) [reply app/modules/adengine-assets.js]: huh, ofc :)

**Comment by fraszczakszymon** [comment]
@leshniak can you revert artifactory changes from package-lock.json?

**Comment by leshniak** [comment]
@fraszczakszymon done

**Review by dmnsobczak — APPROVED** [review]
LGTM. Make sure to (probably manually) reapply some new changes do dev branch since last merge (and fix tests ofc).

**Pull request merged**